### DISCUSSION TITLE
Does including multiple labels impact performance?

### DISCUSSION THREAD

**Discussion by stefanfernandes-mantel** [urchade/GLiNER#180] [Q&A]
Thanks for such a brilliant project. Quick question, does adding more labels to the model impact performance? Or, is there nothing from adding in hundreds of labels if I so wished? Thank you!

**Comment by urchade** [comment]
For the vanilla GLINER model, I recommend a limit of 20 labels. However, for the bi-encoder version, the number of labels is not limited this: You can check this: https://huggingface.co/collections/knowledgator/gliner-bi-encoders-66c492ce224a51c54232657b

**Reply by stefanfernandes-mantel** [reply]
Thanks for the fast response! Does the vanilla model have any inter-label interactions at all? I'm trying to weigh up the pros and cons of running one model with several labels, or multiple models with a small sub-set of labels. Thank you

**Comment by yamiteru** [comment]
I've found out that splitting my labels (50+) into lists of ~5 based on a sub-category has greatly helped with identifying those labels in a text.
